File: unidade-II/pratica-arquivos-2/questao2/questao2.py
import pandas as pd

nAlunos = int(input("Informe a quantidade de alunos: "))

colunas = ["Nome","notas"]
valores = []

notas = []
with open("alunos.txt", "w") as saida:
    for aluno in range(nAlunos):
        nome = input(f"Digite o nome para o aluno {aluno+1}: ")
        for unidade in range(3):
            while True:
                nota = float(input(f"Informe a nota da {unidade+1} unidade: "))
                if nota >= 0 and nota <= 10:                
                    notas.append(nota)
                    break
                else:
                    print("A nota informada deve ser estar no intervalo de 0 a 10.")

        valores.append([nome, notas])
        notas = []

    tabela = pd.DataFrame(columns=colunas, data=valores)
    saida.write(str(tabela))

print()
print(tabela)

File: unidade-II/pratica-arquivos-2/questao2/questao2.c
#include <stdio.h>
#include <stdlib.h>

typedef struct aluno
{
    char nome[21];
    float notas[3];
} Aluno;

int main(void)
{
    int nAlunos;
    printf("Informe a quantidade de alunos: ");
    scanf("%d", &nAlunos);

    Aluno alunos[nAlunos];

    printf("Informe os dados para cada aluno: \n");
    for (int aluno = 0; aluno < nAlunos; aluno++)
    {
        printf("Aluno %d\n", aluno + 1);
        printf("Nome: ");
        scanf(" %[^\n]", alunos[aluno].nome);

        for (int nota = 0; nota < 3; nota++)
        {
            while (1)
            {
                printf("Nota %d: ", nota + 1);
                scanf("%f", &alunos[aluno].notas[nota]);
                if (!(alunos[aluno].notas[nota] >= 0 && alunos[aluno].notas[nota] <= 10))
                {
                    printf("As notas devem estar no intervalo de 0 a 10!\n");
                }
                else
                {
                    break;
                }
            }
        }
    }

    FILE *saida = fopen("notas_alunos.txt", "w");
    if (saida == NULL)
    {
        printf("Erro ao abrir arquivo.\n");
        exit(1);
    }

    fprintf(saida, "Nome:\t\tNotas\n");
    for (int i = 0; i < nAlunos; i++)
    {
        fprintf(saida, "%s\t%2.f\t%2.f\t%2.f\n", alunos[i].nome, alunos[i].notas[0], alunos[i].notas[1], alunos[i].notas[2]);
    }

    fclose(saida);

    printf("Informações gravadas em notas_alunos.txt\n");

    return 0;
}